scanamo/src/main/scala/org/scanamo/DynamoResultStream.scala

```diff
@@ -43,6 +46,27 @@ private[scanamo] trait DynamoResultStream[Req, Res] {
       } yield result
     streamMore(req)
   }
+
+  def streamTo[M[_], T: DynamoFormat](
+    req: Req,
+    pageSize: Int
+  )(implicit M: Alternative[M]): ScanamoOpsT[M, List[Either[DynamoReadError, T]]] = {
+    def streamMore(req: Req, l: Int): ScanamoOpsT[M, List[Either[DynamoReadError, T]]] =
+      exec(withLimit(l)(req)).toFreeT[M].flatMap { res =>
+        val results = items(res).map(ScanamoFree.read[T])
+        if (results.isEmpty)
+          FreeT.liftT(M.empty)
+        else {
+          val l1 = limit(req).map(_ - results.length).fold(pageSize)(Math.min(pageSize, _))
+          lastEvaluatedKey(res)
+            .filterNot(_.isEmpty || l1 <= 0)
```

Comment:
Is the `isEmpty` condition necessary here?

Reply:
@regiskuckaertz ^^^

Reply:
No it shouldn't, let me try.